central/policy/datastore/datastore_impl.go

```diff
@@ -220,12 +220,12 @@ func (ds *datastoreImpl) AddPolicy(ctx context.Context, policy *storage.Policy)
 	if err != nil {
 		return "", errorsPkg.Wrap(err, "getting all policies")
 	}
-	policyNameIDMap := make(map[string]string, len(allPolicies))
+	policyNameToPolicyMap := make(map[string]*storage.Policy, len(allPolicies))
```

Comment:
Does policyNameToPolicyMap have other functions than dedupe the name? If that is the case, we can use string set of names instead?

Reply:
It used to be a map of names (for faster indexing), but I changed it to store full policy because I needed to know if the found policy was a default policy (see line 461)